### ISSUE TITLE
Unexpected invalid position declaration false positive for Vue template

### ISSUE THREAD

**Issue by crystalfp** [stylelint/stylelint#8703]
### What minimal example or steps are needed to reproduce the bug?

```vue
<template>
<table>
      <tr>
        <td style="padding-left: 1rem;">Anything</td>
      </tr>
</table>
</template>
```

### What minimal configuration is needed to reproduce the bug?

```yaml

extends:
  - stylelint-config-standard
  - stylelint-config-standard-vue
  - stylelint-config-standard-scss
  - stylelint-config-recommended-vue/scss
  - stylelint-config-recommended-scss

overrides:
- files:
  - "*.vue"
  - "**/*.vue"
  customSyntax: postcss-html
```

### How did you run Stylelint?

```shell
stylelint --config .stylelintrc.yaml ./src/App.vue
```

### Which Stylelint-related dependencies are you using?

```jaml
    "postcss-html": "^1.8.0",
    "stylelint": "^16.23.0",
    "stylelint-config-recommended-scss": "^15.0.1",
    "stylelint-config-recommended-vue": "^1.6.1",
    "stylelint-config-standard": "^39.0.0",
    "stylelint-config-standard-scss": "^15.0.1",
    "stylelint-config-standard-vue": "^1.0.0",
```

### What did you expect to happen?

No problem to be reported.

### What actually happened?

The following false positive is reported:
```
src/ui/MeasuresCtrl.vue
  225:21  ✖  Unexpected invalid position declaration  no-invalid-position-declaration
```
Note also the error is unrelated to the content.

**Comment by jeddy3** [comment]
@crystalfp Thanks for the report and for using the template.

This should be fixed by https://github.com/stylelint/stylelint/pull/8701.

Closing in favour of the broader https://github.com/stylelint/stylelint/issues/8695 issue.

**Comment by Anoesj** [comment]
EDIT: Posted [here](#9252) as an issue

@jeddy3 Weirdly, I've been running into this problem again lately and I thought it would be due to changes on your end, but in the end I needed to run `pnpm dedupe` to fix these errors. It's possible to have multiple versions of `postcss` installed simultaneously (when `postcss` is a transitive dep of multiple packages and you have pnpm's strict module isolation) and when that happens, things like `instanceof` don't work reliably.

Here's what I think is happening:
- Stylelint's `isInDocument` helper is importing `Node` from postcss and using `instanceof Node`. This can lead to issues when multiple versions of postcss are installed. Another approach would probably be more foolproof.
- Duplicate transitive copies are common with pnpm/yarn. Stylelint shouldn't silently break on it.
- My AI of choice says you can use `.type` here instead of `instanceof Node` and that the `instanceof Node` guard adds nothing once `isDocument(current.document)` is called, because `isDocument` already verifies `node.type === 'document'`. Maybe drop the `instanceof` and guard it with `current.document && typeof current.document === 'object'`? I haven't thoroughly validated this and I can if you'd like me to, but I'm pretty sure you'll instantly know if all this makes sense :smile:
